applications/asset_tracker_v2/src/cloud/cloud_codec/cloud_codec.h

```diff
@@ -255,29 +259,94 @@ enum cloud_codec_event_type {
 };
 
 struct cloud_codec_evt {
+	/** Cloud codec event type */
 	enum cloud_codec_event_type type;
+	/** New config data if type is CLOUD_CODEC_EVT_CONFIG_UPDATE */
 	struct cloud_data_cfg config_update;
 };
 
+/** @brief Event handler prototype.
+ *
+ * @param[in] evt event type
+ */
 typedef void (*cloud_codec_evt_handler_t)(const struct cloud_codec_evt *evt);
 
+/**
+ * @brief Initialize cloud codec.
+ * @details Is called once in the data module to initialize the cloud codec used.
+ * @param[in] cfg initial config data
+ * @param[in] event_handler handler for events coming from the codec
+ *
+ * @note currently only used for config updates in lwm2m
```

Comment:
I'd prefer to have the note after the brief section. To inform the user quickly if there are some special care needed for the API.